src/libraries/position.cairo

```diff
@@ -0,0 +1,142 @@
+use starknet::ContractAddress;
+use yas::numbers::signed_integer::{i32::i32, i64::i64, i128::i128, integer_trait::IntegerTrait};
+use hash::{HashStateTrait, HashStateExTrait};
+use poseidon::PoseidonTrait;
+use poseidon::poseidon_hash_span;
+
+
+// info stored for each user's position
+#[derive(Copy, Drop, Serde, starknet::Store)]
+struct Info {
+    // the amount of liquidity owned by this position
+    liquidity: u128,
+    // fee growth per unit of liquidity as of the last update to liquidity or fees owned
+    fee_growth_inside_0_last_X128: u256,
+    fee_growth_inside_1_last_X128: u256,
+    // the fee owed to the position owner in token0/token1
+    tokens_owed_0: u128,
+    tokens_owed_1: u128,
+}
+
+#[derive(Copy, Drop, Serde, starknet::Store)]
+struct PositionKey {
+    owner: starknet::ContractAddress,
+    tick_lower: i32,
+    tick_upper: i32,
+}
+
+#[starknet::interface]
+trait IPosition<TContractState> {
+    /// @notice Returns the Info struct of a position, given an owner and position boundaries
+    /// @param self The mapping containing all user positions
+    /// @param owner The ContractAddress of the position owner
+    /// @param position_key conformed by the owner and the tick boundaries
+    fn get(self: @TContractState, position_key: PositionKey) -> Info;
+
+
+    fn update(
+        ref self: TContractState,
+        position_key: PositionKey,
+        liquidity_delta: i128,
+        fee_growth_inside_0_X128: u256,
+        fee_growth_inside_1_X128: u256
+    );
+}
+
+fn generate_hashed_position_key(key: @PositionKey) -> felt252 {
+    let mut serialized: Array<felt252> = ArrayTrait::new();
```

Comment:
```suggestion
    let mut serialized = array![];
```

Reply:
With the new way to Hash struct this is no longer necessary. This comment was useful anyway because I wasn't aware of this trick. Thanks!